streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -780,4 +782,17 @@ public void shouldNotAllowToAddGlobalStoreWithSourceNameEqualsProcessorName() {
             sameNameForSourceAndProcessor,
             new MockProcessorSupplier());
     }
+
+    @Test
+    public void shouldAllowAddGlobalStoreWithNonKVStore() {
+        final StoreBuilder windowStoreBuilder = new MockWindowStoreBuilder("store", false);
+        builder.addGlobalStore((StoreBuilder<WindowStore>) windowStoreBuilder.withLoggingDisabled(),
```

Comment:
Does the cast make sense? `addGlobalStore` has not generic type any more and thus the type cannot be checked anyway?

Reply:
I guess you are right. The cast was useful when `addGlobalStore` still had a generic type, as it would refuse to compile. I should have removed it.

Reply:
@mjsax fixed it in 24f1fc3